src/dsp_tools/commands/xml_validate/api_connection.py

```diff
@@ -0,0 +1,75 @@
+from dataclasses import dataclass
+from typing import Any
+from typing import cast
+
+import requests
+from loguru import logger
+from requests import ReadTimeout
+from requests import RequestException
+from requests import Response
+
+from dsp_tools.models.exceptions import InternalError
+from dsp_tools.models.exceptions import UserError
+
+
+@dataclass
+class OntologyConnection:
+    api_url: str
+    shortcode: str
+
+    def _get(self, url: str, headers: dict[str, Any] | None = None) -> Response:
+        """
+        Sends a get request to the designated url
+
+        Args:
+            url: URL for the request
+            headers: headers for the request
+
+        Returns:
+            Response of the request if it was a 200 response code
+
+        Raises:
+            InternalError: in case of errors raised by the requests library
+            UserError: If a non-200 response was given
+        """
+        try:
+            response = requests.get(url, headers=headers, timeout=100)
```

Comment:
One thing I completely forgot in my yesterday's review: For every new client that we introduce, we shouldn't forget to add logging statements for every request and every response. As an example how to do it:

https://github.com/dasch-swiss/dsp-tools/blob/e65c75315da8039fa9533ca3b79e3a4802768cd8/src/dsp_tools/commands/ingest_xmlupload/bulk_ingest_client.py#L71